### PR TITLE
Add SmoothedUniformFactory

### PR DESCRIPTION
**Pull request by jschueller**
first guess by method of moments + MLE

also fixed SmoothedUniform:
- ddf
- computeMinimumVolumeLevelSetWithThreshold(relying on getSamplebyQMC shadowed by RandomMixture)
- better graph (the old one looked like Normal with the default parametrization)

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/SmoothedUniform.cxx [hunk 1]
```diff
@@ -124,12 +124,20 @@ Sample SmoothedUniform::getSample(const UnsignedInteger size) const
   return result;
 }
 
+Sample SmoothedUniform::getSampleByQMC(const UnsignedInteger size) const
+{
+  SampleImplementation result(size, 1);
+  const Point u(SobolSequence(1).generate(size).getImplementation()->getData());
+  const Scalar ba = b_ - a_;
+  for (UnsignedInteger i = 0; i < size; ++i)
+    result(i, 0) = a_ + ba * u[i] + sigma_ * DistFunc::rNormal();
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/SmoothedUniform.cxx:133]: ??? getSampleByQMC *should be* a CDF inversion based on a unique Sobol sequence!

Comment (jschueller) [reply lib/src/Uncertainty/Distribution/SmoothedUniform.cxx:133]: oh, right, there is no static keyword

**Review by regislebrun — APPROVED** [review]

File: python/src/SmoothedUniformFactory_doc.i.in [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+The parameters are estimated by method of moments:
+
+.. math::
+    :nowrap:
+
+    \begin{eqnarray*}
+      \displaystyle\delta=\sqrt[4]{120 * \hat{\sigma}^4 * (3-\hat{\kappa})}\\
+      \displaystyle\Hat{a}_n=\hat{\mu} - \frac{\delta}{2}\\
+      \displaystyle\Hat{b}_n=\hat{\mu} + \frac{\delta}{2}\\
+      \displaystyle\Hat{sigma}_n=\sqrt{\hat{\sigma}^2 - \frac{\delta^2}{12}}
```
Comment (josephmure) [inline python/src/SmoothedUniformFactory_doc.i.in:19]:
```suggestion
      \displaystyle\Hat{\sigma}_n=\sqrt{\hat{\sigma}^2 - \frac{\delta^2}{12}}
```

File: python/src/SmoothedUniformFactory_doc.i.in [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+----------
+sample : 2-d sequence of float
+    Data.
+param : sequence of float
+    The parameters of the :class:`~openturns.SmoothedUniform`.
+
+Returns
+-------
+dist : :class:`~openturns.SmoothedUniform`
+    The estimated distribution as a :class:`openturns.SmoothedUniform`.
```
Comment (josephmure) [inline python/src/SmoothedUniformFactory_doc.i.in:54]:
```suggestion
    The estimated distribution as a :class:`~openturns.SmoothedUniform`.
```

File: python/src/SmoothedUniformFactory_doc.i.in [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    
+    In the first usage, the default :class:`openturns.SmoothedUniform` distribution is built."
```
Comment (josephmure) [inline python/src/SmoothedUniformFactory_doc.i.in:56]:
```suggestion
    In the first usage, the default :class:`~openturns.SmoothedUniform` distribution is built."
```

File: python/test/t_SmoothedUniformFactory_std.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+estimatedDistribution = factory.build()
+print("Default distribution=", estimatedDistribution)
+estimatedDistribution = factory.build(distribution.getParameter())
+print("Distribution from parameters=", estimatedDistribution)
+estimatedSmoothedUniform = factory.buildAsSmoothedUniform(sample)
+print("SmoothedUniform          =", distribution)
+print("Estimated uniform=", estimatedSmoothedUniform)
+estimatedSmoothedUniform = factory.buildAsSmoothedUniform()
+print("Default uniform=", estimatedSmoothedUniform)
+estimatedSmoothedUniform = factory.buildAsSmoothedUniform(distribution.getParameter())
```
Comment (josephmure) [inline python/test/t_SmoothedUniformFactory_std.py:26]: This is not actually tested due to the IGNOREOUT. Or is it just to see if the code doesn't crash?

**Review by josephmure — CHANGES_REQUESTED** [review]

File: python/test/t_SmoothedUniformFactory_std.py (hunk 5 above)
Comment (jschueller) [reply python/test/t_SmoothedUniformFactory_std.py:26]: yes, I could add some asserts

**Comment by jschueller** [comment]
@josephmure can I merge this one too ? some commits are bug fixes that can be included in 1.22

**Review by josephmure — APPROVED** [review]

**Pull request merged**